File: poc-core/src/main/java/poc/core/service/OrganizationService.java
package poc.core.service;


import java.util.List;

import poc.core.domain.Organization;


public interface OrganizationService {

  List<Organization> getOrganizations();

  Organization getOrganization(Long id);

  Organization saveOrganization(Organization organization);

  Organization deleteOrganization(Organization organization);

}

File: poc-web-gui-angular/poc-angular/src/app/organizations/organization.service.ts
import {Injectable} from '@angular/core';

import {EntityService} from 'entity-lib';
import {NGXLogger} from 'ngx-logger';
import {Organization} from '../core/domain';
import {PocApiService} from '../core/service';
import {organizationMeta} from './organization-meta';

@Injectable({
  providedIn: 'root'
})
export class OrganizationService extends EntityService<Organization> {

  constructor(
    public apiService: PocApiService,
    public logger: NGXLogger) {
    super(organizationMeta, apiService, logger);
  }
}
